src/core/EdFi.DataManagementService.Core/Middleware/DuplicatePropertiesMiddleware.cs

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using EdFi.DataManagementService.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+using Microsoft.Extensions.Logging;
+using static EdFi.DataManagementService.Core.Response.FailureResponse;
+
+namespace EdFi.DataManagementService.Core.Middleware;
+
+/// Parse did not find errors in repeated values, it is identified until an attempt is made to use the JsonNode
+/// Please see https://github.com/dotnet/runtime/issues/70604 for information on the JsonNode bug this code is working-around.
+internal class DuplicatePropertiesMiddleware(ILogger logger) : IPipelineStep
+{
+    private static readonly JsonSerializerOptions _serializerOptions =
+        new() { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping };
+    private const string TestForDuplicateObjectKeyWorkaround = "Test";
+    private const string Pattern = @"Key: (.*?) \((.*?)\)\.(.*?)$";
+
+    public async Task Execute(PipelineContext context, Func<Task> next)
+    {
+        logger.LogDebug(
+            "Entering DuplicatePropertiesMiddleware - {TraceId}",
+            context.FrontendRequest.TraceId
+        );
+
+        if (context.FrontendRequest.Body != null)
+        {
+            try
+            {
+                JsonNode? node = JsonNode.Parse(context.FrontendRequest.Body);
+
+                if (node is JsonObject jsonObject)
+                {
+                    _ = node[TestForDuplicateObjectKeyWorkaround];
```

Comment:
This will identify if the problem is at the first level.

Reply:
Please add your comment here as a comment in the code